### PR TITLE
SEAR-91-No search results criteria error message

### PR DESCRIPTION
**Pull request by Nasir2712**
#### Which User story does this relate to (link)?
- https://osi-cwds.atlassian.net/browse/SEAR-91

#### Brief feature description
-show a different message when a user selects no search criteria.

#### Does this PR block/effect other stories?
- no

#### Do any commands needs to be run in order to run the project with this PR?
-no

#### Code coverage percent diff as a result of this PR?
- n/a

### CODE REVIEW

File: app/javascript/reducers/searchReducer.js [hunk 1]
```diff
@@ -19,27 +20,30 @@ const initialState = {
   errors: {},
   countyTypes: [],
   facilityTypes: [],
-  user: {
-    county_code: ''
-  }
+  userCounty: '',
```
[the remark this answers is not in the document]
Comment (Nasir2712) [reply app/javascript/reducers/searchReducer.js]: it's a string.

**Comment by Nasir2712** [comment]
@ratneshraval addressed all comments.

File: app/javascript/sagas/fetchFacilityByIdSaga.js [hunk 2]
```diff
@@ -10,9 +10,13 @@ export function * fetchFacilityById (action) {
     const url = '/facilities/facility'
     const response = yield call(fetchRequest, url, 'POST', action.payload.facilityParams)
     const data = yield call([response, response.json])
-    yield put(fetchSuccess({facility: data.facility, children: data.children, complaints: data.complaints}))
+    if (response.status >= 200 && response.status < 300) {
```
Comment (ratneshraval) [inline app/javascript/sagas/fetchFacilityByIdSaga.js]: Let's handle the response success/failure in different method. Something like `isSuccess` or `isResponseSuccessful`

File: app/javascript/actions/searchActions.js [hunk 3]
```diff
@@ -2,12 +2,16 @@ export const SEARCH_DICTIONARIES_FETCH = 'SEARCH_DICTIONARIES/FETCH'
...
 export const SEARCH_RESULTS_FETCH_COMPLETE = 'SEARCH_RESULTS/FETCH_COMPLETE'
+export const SEARCH_RESULTS_FETCH_ERROR = 'SEARCH_RESULTS/FETCH_ERROR'
+export const NO_SEARCH_CRITERIA_ERROR = 'SEARCH_RESULTS/FETCH_NOTHING_ERROR'
 export const HANDLE_INPUT_CHANGE = 'HANDLE_INPUT_CHANGE'
 export const HANDLE_TOGGLE = 'HANDLE_TOGGLE'
 export const HANDLE_RESET_FORM = 'HANDLE_RESET_FORM'
 export const HANDLE_DROPDOWN_CHANGE = 'HANDLE_DROPDOWN_CHANGE'
 export const HANDLE_PAGENUMBER_CHANGE = 'HANDLE_PAGENUMBER_CHANGE'
 export const HANDLE_DROPDOWN_AND_PAGENUMBER_CHANGE = 'HANDLE_DROPDOWN_AND_PAGENUMBER_CHANGE'
+export const NO_RESULTS_ERROR_MESSAGE = 'No results were found with the selected search criteria, please refine your search criteria and try again.'
```
Comment (ratneshraval) [inline app/javascript/actions/searchActions.js]: Error message constants with action constants :( , makes it little confusing

Comment (sunilb0575) [reply app/javascript/actions/searchActions.js]: I have added actionTypes.js along with containers for facility profile, as soon as I solve failing test cases I will create a PR.

**Comment by Nasir2712** [comment]
@ratneshraval check out my second commit for refactoring fetchRequest in searchResultsSaga.

**Pull request merged**